engine-standalone-storage/src/sync/types.rs

```diff
@@ -133,7 +133,7 @@ pub enum TransactionKind {
     FactorySetWNearAddress(Address),
     FundXccSubAccount(FundXccArgs),
     /// Silo operations
-    SetFixedGasCost(silo::parameters::FixedGasCostArgs),
+    SetSiloParams(Option<silo::parameters::SiloParamsArgs>),
```

Comment:
This is technically a breaking change because the `set_fixed_gas_cost` is being removed from the list of known transactions. This means we will no longer be able to index old transactions calling that function using the Borealis infrastructure (refiner/engine). This might be ok if we will never need to index old transactions because the first production silo is not yet deployed. But I wanted to call it out in case this breaking change might cause a problem for any demos we have set up.

Reply:
We can return the `set_fixed_gas_cost` back but with an extra check for `erc20_fallback_address != None` if the passed  `fixed_gas_cost != None`. and set both values as None if the passed `fixed_gas_cost` is None.

Reply:
Sounds good to me.

Reply:
Here is the implementation https://github.com/aurora-is-near/aurora-engine/pull/765/commits/2ea368ceed5f1440b5f57702c5df19c63ed13683.

For the existing demo, the admin needs to call the `set_silo_params` to be able to call the `set_fixed_gas_cost` again.

Reply:
Yes, to be backwards compatible (able to index all previous transactions) we must include all past and present top-level contract functions in `TransactionKind`.

Reply:
Reverted in bd13bf876a88b80552fb961e29a0a12e26919c8d

Reply:
@birchmd @aleksuss reverted in bd13bf876a88b80552fb961e29a0a12e26919c8d